### PR TITLE
Launchpad: Add Newsletter Post-Setup Screen

### PR DESCRIPTION
**Pull request by edanzer**
### Proposed Changes

**Addresses this issue:** https://github.com/Automattic/wp-calypso/issues/67974

**Requires this backend code:** D88664-code

We want to allow users in the Newsletter tailored flow to return from Launchpad the Newsletter Setup screen. In line with past conversations, rather than send users back to the original Setup screen, we are creating a new, Newsletter Post Setup stepper flow and screen. Users can navigate between Launchpad and the Post Setup screen, and changes made on new screen should be reflected on their site, including the Launchpad website preview. 

Specific changes include: 
* Create Newsletter Post Setup stepper flow
* Create Newsletter Post Setup step/screen within the new flow
* Ensure Newsletter Post Setup screen pre-populates site title/tagline/icon/color from the Redux SITE_STORE or API.
* Ensure changes to site title/tagline/icon/color are updated when saving/continuing on from screen.
* Add link to new flow/screen from Launchpad. 
* In separate backend PR (link above): Updated global-styles-variation to get/set site accent color 

### Testing Instructions

1. Checkout this branch, and run yarn and yarn start if needed.
2. Apply D88664-code to your sandbox. 
2. Start a new Newsletter site from https://wordpress.com/hp-2022-tailored-flows/
3. Proceed through flow until you arrive at Launchpad. 
4. When you arrive at Launchpad, replace `https://worpdress.com` with `http://calypso.localhost:3000/` to ensure you are seeing local branch.
5. TEST: From Launchpad, confirm the 'Personalize Newsletter' task is active/clickable. 
6. TEST: Click the 'Personalize Newsletter' task. You should arrive at a screen that looks exactly like the original setup screen, but the url should be `http://calypso.localhost:3000/setup/newsletterPostSetup?flow=newsletter-post-setup&siteSlug=[YOUR_SITE_SLUG]`
7. TEST: Confirm you can edit the site title, tagline, icon, and color. 
8. TEST: Click 'Continue' and confirm you return to Launchpad. 
9. TEST: Confirm the Launchpad web preview shows your updated site title, tagline, icon, and color.

### Screenshots

For potential translations and for reference during testing, the new Newsletter Post Setup screen looks like this.

newsletter-post-setup

### CODE REVIEW

**Review by sharpirate — COMMENTED** [review]
This looks good and tests well for me!

✅: From Launchpad, confirm the 'Personalize Newsletter' task is active/clickable.
✅: Click the 'Personalize Newsletter' task. You should arrive at a screen that looks exactly like the original setup screen, but the url should be http://calypso.localhost:3000/setup/newsletterPostSetup?flow=newsletter-post-setup&siteSlug=[YOUR_SITE_SLUG]
✅: Confirm you can edit the site title, tagline, icon, and color.
✅: Click 'Continue' and confirm you return to Launchpad.
✅: Confirm the Launchpad web preview shows your updated site title, tagline, icon, and color.

I did notice a ton of requests every time I change the color on the color picker, also a bunch of duplicate requests when I first land on the Personalize page:

image

I am assuming this has to do with the caching issue you mentioned (I wasn't able to follow through those discussions, I guess Matt's video recording will provide more context here)

I am good with the changes so far but I will hold off approving for now due to the pending caching disctussion

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx
Comment (agrullon95) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx]: Is this still needed since we merged D89679-code?

File: client/landing/stepper/declarative-flow/newsletter-post-setup.ts
Comment (agrullon95) [inline client/landing/stepper/declarative-flow/newsletter-post-setup.ts]: I think we can remove `clearCache=true` query param now since D89679-code has been merged.

**Comment by agrullon95** [comment]
Tested with new newsletter site:

- [x] TEST: From Launchpad, confirm the 'Personalize Newsletter' task is active/clickable.
- [x] TEST: Click the 'Personalize Newsletter' task. You should arrive at a screen that looks exactly like the original setup screen, but the url should be http://calypso.localhost:3000/setup/newsletterPostSetup?flow=newsletter-post-setup&siteSlug=[YOUR_SITE_SLUG]
- [x] TEST: Confirm you can edit the site title, tagline, icon, and color.
- [x] TEST: Click 'Continue' and confirm you return to Launchpad.
- [x] TEST: Confirm the Launchpad web preview shows your updated site title, tagline, icon, and color.

I tested on an existing blog with a random theme and noticed that the fetch site accent color endpoint (`/sites/${ ID }/global-styles-variation/site-accent-color`) returns a `null` color.

I also noticed that the fetch site account color endpoint (`/sites/${ ID }/global-styles-variation/site-accent-color`) is called multiple times (~5 times) when I load the newsletter-post-setup screen.

File: packages/onboarding/src/utils/flows.ts
Comment (sharpirate) [inline packages/onboarding/src/utils/flows.ts:2]: should this be 'newsletter-post-setup'

**Comment by sharpirate** [comment]
- [x] TEST: From Launchpad, confirm the 'Personalize Newsletter' task is active/clickable.
- [x] TEST: Click the 'Personalize Newsletter' task. You should arrive at a screen that looks exactly like the original setup screen, but the url should be http://calypso.localhost:3000/setup/newsletterPostSetup?flow=newsletter-post-setup&siteSlug=[YOUR_SITE_SLUG]
- [x] TEST: Confirm you can edit the site title, tagline, icon, and color.
- [x] TEST: Click 'Continue' and confirm you return to Launchpad.
- [x] TEST: Confirm the Launchpad web preview shows your updated site title, tagline, icon, and color.

I am seeing the multiple requests too as Anthony mentioned

**Review by Addison-Stavlo — COMMENTED** [review]
I tested from the launchpad and this all worked as expected:

TEST: From Launchpad, confirm the 'Personalize Newsletter' task is active/clickable.
TEST: Click the 'Personalize Newsletter' task. You should arrive at a screen that looks exactly like the original setup screen, but the url should be http://calypso.localhost:3000/setup/newsletterPostSetup?flow=newsletter-post-setup&siteSlug=[YOUR_SITE_SLUG]
TEST: Confirm you can edit the site title, tagline, icon, and color.
TEST: Click 'Continue' and confirm you return to Launchpad.
TEST: Confirm the Launchpad web preview shows your updated site title, tagline, icon, and color.

I also tested from the initial start flows setup form and dont see any regressions there.

I dont see any issues/Qs on the code side besides what has already been mentioned.

>I tested on an existing blog with a random theme and noticed that the fetch site accent color endpoint (/sites/${ ID }/global-styles-variation/site-accent-color) returns a null color.
> I also noticed that the fetch site account color endpoint (/sites/${ ID }/global-styles-variation/site-accent-color) is called multiple times (~5 times) when I load the newsletter-post-setup screen.

I wonder if this is why I see a bit of a delay in items updating when I visit the post-setup form.  I realize my color picks for the GIF below weren't the best (dark green vs dark blue)... but you can see that it loads in with the dark blue initially, and takes a few to update to the actual value of the accent color (longer than the other items on the page).  That said, I imagine we could address these concerns in a follow up if it feels more manageable that way.

color updates later?

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx
Comment (edanzer) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx]: Correct, this is no longer needed. I had just been waiting for that diff to be merged. It's been removed now.

File: client/landing/stepper/declarative-flow/newsletter-post-setup.ts
Comment (edanzer) [reply client/landing/stepper/declarative-flow/newsletter-post-setup.ts]: Correct. This has been removed now.

File: packages/onboarding/src/utils/flows.ts
Comment (edanzer) [reply packages/onboarding/src/utils/flows.ts:2]: @Sharpirate - While this is technically a separate flow, I think we still want to treat it as part of the newsletter flow as far as this method is concerned. This is essentially like a sub-flow for newsletters.

**Comment by mattwiebe** [comment]
So I'm not seeing this function the way it should. When going through signup I set my color to this orange-y color:

Screen Shot 2022-10-13 at 11 55 25

Then I click "Personalize Newsletter" and pick a greenish color and Continue:

Screen Shot 2022-10-13 at 11 56 51

And I'm just back to my orange-y color. It looks like c7eb185653cfe0dbd8b2e7a6d19413ebf4e8fb4b is trying to get the color from a `color` query param that isn't present?

**Comment by jeyip** [comment]
## Testing
### Requirements

- [x] From Launchpad, confirm the 'Personalize Newsletter' task is active/clickable.
- [x] Click the 'Personalize Newsletter' task. You should arrive at a screen that looks exactly like the original setup screen, but the url should be http://calypso.localhost:3000/setup/newsletterPostSetup?flow=newsletter-post-setup&siteSlug=[YOUR_SITE_SLUG]
- [ ] Confirm you can edit the site title, tagline, icon, and color.
- [x] Click 'Continue' and confirm you return to Launchpad.
- [x] Confirm the Launchpad web preview shows your updated site title, tagline, icon, and color.

### Browsers
[x] Chrome

The personalize step doesn't let me update the site icon if I already set the icon during the first pass through the newsletter onboarding flow.

File: client/landing/stepper/hooks/use-save-accent-color.ts
Comment (jeyip) [inline client/landing/stepper/hooks/use-save-accent-color.ts:5]: Post, patch, and delete actions are typically handled through redux / core-data. It seems like it would be useful here to update any related state in the redux / core-data stores easily, which, from what I can tell, we're not currently doing.

Ex. https://github.com/Automattic/wp-calypso/blob/trunk/packages/data-stores/src/site/actions.ts#L51

Is there a reason we're switching up this convention and using a custom hook to return this request method?

**Review by mattwiebe — APPROVED** [review]
Aha! I was having `git` troubles and got this working now. Seems to work properly, looks good now after getting the right code running. I know that @jeyip was having trouble with icon uploading? but everything checks out now for me.

**Comment by jeyip** [comment]
> I know that @jeyip was having trouble with icon uploading? but everything checks out now for me.

I think things are fine -- more context here p1666028542342679-slack-CHN6J22MP. If what I've investigated make sense to you Erick, +1 from me

**Comment by Addison-Stavlo** [comment]
> The personalize step doesn't let me update the site icon if I already set the icon during the first pass through the newsletter onboarding flow.

Ah, I think I see this now as well. It seems that if I click on the existing logo image itself it wont work, but if I click "replace" link underneath it will open the file browser and let me update an image. Perhaps an oversight from the initial development of this component since it wasn't initially intended to handle the case of an image already selected. This could be good to create a follow-up issue and improvement for.

**Review by jeyip — APPROVED** [review]

**Comment by jeyip** [comment]
> Perhaps an oversight from the initial development of this component since it wasn't initially intended to handle the case of an image already selected. This could be good to create a follow-up issue and improvement for.

Agree with these thoughts and a follow-up

File: client/landing/stepper/hooks/use-save-accent-color.ts
Comment (edanzer) [reply client/landing/stepper/hooks/use-save-accent-color.ts:5]: @jeyip - Site accent color is something we store or update in state currently. It's a bit of an odd value in that a) its part of theme json, and even our changes operate directly on theme json and b) it's very theme specific (ie, the endpoint only works and updates theme json if one of two themes are active). Given all that, I figured it wasn't worth trying to add this to redux state with the associated actions/selectors/dispatch configuration. 

That said, my understanding of when/how things should be updated via redux vs directly is incomplete. If you think we should look at adjusting this, I'm happy to do so.

Comment (jeyip) [reply client/landing/stepper/hooks/use-save-accent-color.ts:5]: This is something we can definitely chat about, but feel free to merge the PR for now. We can refactor in a follow-up if we decide that it's worthwhile.

**Pull request merged**
